ext/pdo_dblib/dblib_driver.c

```diff
@@ -143,60 +143,92 @@ static zend_long dblib_handle_doer(pdo_dbh_t *dbh, const char *sql, size_t sql_l
 	return DBCOUNT(H->link);
 }
 
-static int dblib_handle_quoter(pdo_dbh_t *dbh, const char *unquoted, size_t unquotedlen, char **quoted, size_t *quotedlen, enum pdo_param_type paramtype)
+static int dblib_handle_quoter(pdo_dbh_t *dbh, const char *unquoted, size_t unquoted_len, char **quoted, size_t *quoted_len, enum pdo_param_type param_type)
 {
+	const char *hex = "0123456789abcdef";
+	char *q;
+	*quoted_len = 0;
+	char is_ascii = 1;
+	char is_int = 1;
+
+	switch (PDO_PARAM_TYPE(param_type)) {
+		case PDO_PARAM_LOB:
+			/*
+			 * Binary safe quoting
+			 */
+			*quoted_len += (unquoted_len * 2) + 2; // 2 chars per byte +2 for "0x" prefix
+			q = *quoted = safe_emalloc(2, *quoted_len, 3);
+
+			*q++ = '0';
+			*q++ = 'x';
+			for (size_t i = 0; i < unquoted_len; i++) {
+				*q++ = hex[(*unquoted >> 4) & 0xF];
+				*q++ = hex[(*unquoted++) & 0xF];
+			}
+			break;
 
-	int useBinaryEncoding = 0;
-	const char * hex = "0123456789abcdef";
-	size_t i;
-	char * q;
-	*quotedlen = 0;
 
-	/*
-	 * Detect quoted length and if we should use binary encoding
-	 */
-	for(i=0;i<unquotedlen;i++) {
-		if( 32 > unquoted[i] || 127 < unquoted[i] ) {
-			useBinaryEncoding = 1;
+		case PDO_PARAM_BOOL:
+			*quoted_len = 3;
+			if (!unquoted_len // empty strings treated as FALSE anyway
+				|| (unquoted_len == 1 && (*unquoted == '0' || *unquoted == ' ')) // one-char strings depends on a value
```

Comment:
Why does a space count as false but two spaces don't? It would be cleaner To say that empty strings and '0' are false, everything else true. But I wonder if there's a case here I'm not thinking of.